nix/shell.nix

```diff
@@ -23,7 +23,7 @@ in mkShell {
       # build specific utilities
       clojure maven watchman
       # lint specific utilities
-      clj-kondo zprint
+      clj-kondo zprint clojure-lsp
```

Comment:
The `lint-fix` target uses the `clojure` shell, so we should probably just put this in there. Honestly, the other 2 utilities here should probably also be in the `clojure` shell only:
https://github.com/status-im/status-mobile/blob/fc284f466fff16790fbd6c2585f54f2bb1209c54/nix/shells.nix#L19-L29